### PR TITLE
Update Templates and Generic Programming

### PR DESCRIPTION
**Pull request by gjxo12**
While reading the documentation for that part, I checked once to see if it was the same as the original.

 Not all content has been completely corrected, but please review.

### CODE REVIEW

File: sections/Templates.md [hunk 1]
```diff
@@ -319,8 +361,7 @@ STL은 이 접근법을 사용한다.
 
 ## <a name="SS-concepts"></a>T.concepts: 컨셉 규칙들
 
-컨셉(Concept)은 템플릿 인자들에 대한 요구사항을 지정하기 위한 기능이다.
-컨셉은 [ISO technical specification](#Ref-conceptsTS)이지만, 현재는 GCC만 이를 지원하고 있다.  
+컨셉(Concept)은 템플릿 인자들에 대한 요구사항을 지정하기 위한 C++ 20 기능이다.
```
Comment (utilForever) [inline sections/Templates.md]:
```suggestion
컨셉(Concepts)은 템플릿 인자들에 대한 요구사항을 지정하기 위한 C++20 기능이다.
```

File: sections/Templates.md [hunk 2]
```diff
@@ -515,7 +556,7 @@ TC++PL 4, Palo Alto TR, Sutton
 
 하나의 클래스나 알고리즘을 문법적으로 제약하는 것은 컨셉이 의도하는 바가 아니며, 컨셉의 메커니즘을 적용했을 때의 효율을 완전히 끌어낼 수 없다.
 
-분명, 컨셉을 정의하는 것은 컨셉 구현을 사용할 수 있는 (예컨대 GCC 6.1 이후의 버전으로 컴파일 되는) 코드에 유용할 것이다.
+분명, 컨셉을 정의하는 것은 컨셉 구현을 사용할 수 있는 (예컨대 c++20 혹은 그 이상) 코드에 유용할 것이다.
```
Comment (utilForever) [inline sections/Templates.md]:
```suggestion
분명, 컨셉을 정의하는 것은 구현을 사용할 수 있는 (예를 들어, C++20 또는 그 이후) 코드에 유용할 것이다.
```

File: sections/Templates.md [hunk 3]
```diff
@@ -1429,10 +1470,13 @@ C++ 17 에서는 이 규칙처럼 템플릿 인자들을 생성자의 실행 인
 해당 문서에 따르면:
 "The acronym SCARY describes assignments and initializations that are Seemingly erroneous (appearing Constrained by conflicting generic parameters), but Actually work with the Right implementation (unconstrained bY the conflict due to minimized dependencies."
 
+##### Note
+이는 템플릿 매개변수에 의존하지 않는 람다에도 적용된다
 ##### Enforcement
```
Comment (utilForever) [inline sections/Templates.md]:
```suggestion
##### Note

템플릿 매개변수에 의존하지 않는 람다에도 적용된다.

##### Enforcement
```

File: sections/Templates.md [hunk 4]
```diff
@@ -1429,10 +1470,13 @@ C++ 17 에서는 이 규칙처럼 템플릿 인자들을 생성자의 실행 인
...
 
 * 멤버 타입이 의존하지 않는 템플릿 매개변수가 있다면 지적한다
 * 멤버 함수가 의존하지 않는 템플릿 매개변수가 있다면 지적한다
+* 람다가 의존하지 않는 템플릿 매개변수가 있다면 지적한다
```
Comment (utilForever) [inline sections/Templates.md]:
```suggestion
* 모든 템플릿 매개 변수에 의존하지 않는 멤버 타입을 표시하라.
* 모든 템플릿 매개 변수에 의존하지 않는 멤버 함수를 표시하라.
* 모든 템플릿 매개 변수에 의존하지 않는 람다나 변수 템플릿을 표시하라.
```

**Review by utilForever — CHANGES_REQUESTED** [review]
Can you check my comments? Thanks! 🔍

**Comment by gjxo12** [comment]
Thank you for your review.

I checked all commit.

Thanks for pointing out my mistake.

**Review by utilForever — APPROVED** [review]
I checked the changes. Thanks! 🍹

**Review by GiyeonBang — CHANGES_REQUESTED** [review]
Idiomatically, `컨셉` is used more common, but according to the National Institute of Korean Language(국립국어원), it is correct to write `컨셉` as `콘셉트`.
Or, since it is a proper noun, it is recommended not to describe it in Korean.
PS: I ask for opinions from other reviewers.

**Review by GiyeonBang — CHANGES_REQUESTED** [review]
Same as above.
`컨셉(Concepts)` --> `콘셉츠(Concepts)`
Or just `Concepts`

**Review by luncliff — APPROVED** [review]
Sorry for the late check. 

Would you check https://github.com/CppKorea/CppCoreGuidelines/pull/162#pullrequestreview-1485292285
?

Because `C++ 20 Concepts` is a unique name for the language feature.
I vote for just writing `Concepts` without Korean words.
But I respect the contributors' choice. Go ahead with your opinion.

**Comment by gjxo12** [comment]
I think just writing `Concepsts` without Korean words.

This expression looks concise and readable.

If the reviewers don't have any problems, I'd suggest modifing them with `Concepts`

By any chance, if I modify it, can I go through the process of editing -> commit -> push again?

**Comment by utilForever** [comment]
> I think just writing `Concepsts` without Korean words.
> 
> This expression looks concise and readable.
> 
> If the reviewers don't have any problems, I'd suggest modifing them with `Concepts`
> 
> By any chance, if I modify it, can I go through the process of editing -> commit -> push again?

I agree. You can commit to this branch before PR is merged. 🗡️

**Comment by gjxo12** [comment]
> > I think just writing `Concepsts` without Korean words.
> > This expression looks concise and readable.
> > If the reviewers don't have any problems, I'd suggest modifing them with `Concepts`
> > By any chance, if I modify it, can I go through the process of editing -> commit -> push again?
> 
> I agree. You can commit to this branch before PR is merged. 🗡️

Ok. I refactored all words of "컨셉" to "Concepts".
If there are any other issues, i'll fix them again.

**Review by GiyeonBang — APPROVED** [review]

**Pull request merged**